File: src/main/java/raccoonlang/tokenizer/TokenMismatchException.java
package raccoonlang.tokenizer;

public class TokenMismatchException extends RuntimeException {
    public TokenMismatchException(Token token, TokenType tokenType) {
        //TODO: better exception message
        super("Expected '" + tokenType + "' but got '" + token.getTokenType() + "'");
    }
}

File: Tokenizing/Exceptions/TokenMismatchException.cs
namespace Raccoonlang.Tokenizing.Exceptions;

using System;

public class TokenMismatchException : Exception
{
    public TokenMismatchException(Token token, TokenType type)
    : base($"{token.FileName}:{token.Line}:{token.Column} | Expected `{type}` but got `{token.Type}` instead.") {}
}